### PR TITLE
ORC-1613: Zstd decompression supports direct buffer

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
`ZstdCodec` implements the `DirectDecompressionCodec` interface.

### Why are the changes needed?
`zstd-jni` supports direct buffer decompression, which can reduce Buffer copying.

### How was this patch tested?
add UT

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/ZstdCodec.java [hunk 1]
```diff
@@ -214,6 +214,11 @@ public boolean compress(ByteBuffer in, ByteBuffer out,
 
   @Override
   public void decompress(ByteBuffer in, ByteBuffer out) throws IOException {
+    if (in.isDirect() && out.isDirect()) {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/ZstdCodec.java:217]: Just a question. When is this used in Apache ORC code base?

Comment (cxzl25) [reply java/core/src/java/org/apache/orc/impl/ZstdCodec.java:217]: If we use zerocopy `orc.use.zerocopy=true`, we may use the direct buffer, which can fallback to the original way of reading if the codec doesn't implement `DirectDecompressionCodec`.

https://github.com/apache/orc/blob/4eff23aa3ff6180f00cab472b91c3f67f82d6683/java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java#L600-L607

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/ZstdCodec.java:217]: Got it.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to main/2.0. Thank you, @cxzl25 .

**Pull request closed without merging**